frontend/src/pages/user/UserAnswerPage.js

```diff
@@ -39,9 +42,46 @@ export default function UserAnswerPage() {
   const lastPage =
     questions && Array.isArray(questions) ? questions.length + 1 : 0;
 
+  const totalQuestion =
+    questions && Array.isArray(questions) ? questions.length : 0;
+  const passingScore = 75;
+
+  useEffect(() => {
+    setFailed(scorePercentage < passingScore ? true : false);
+    setPassed(scorePercentage >= passingScore ? true : false);
+  }, [scorePercentage]);
```

Comment:
Remove this useEffect, it is taking a memory. Just add the logic if fail or pass after api response.
Note that only use 1 variable. setIsPassed with boolean value